js/controllers/AutocompleteHandler.js

```diff
@@ -182,10 +184,15 @@
 	$.pkp.controllers.AutocompleteHandler.prototype.fetchAutocomplete =
 			function(callingElement, request, response) {
 		var $textInput;
+		var self = this;
 
 		$textInput = this.textInput;
 		$textInput.addClass('spinner');
-		$.post(this.getAutocompleteUrl(), { term: request.term }, function(data) {
+		if (self.lastRequest_) {
+			self.lastRequest_.abort('Aborted due to new user request');
```

Comment:
I don't think this will actually do anything to improve the server performance. This will prevent jQuery from firing the callback when the request is returned, but the server will still go through processing it.

Our new UI components use a couple of strategies to minimize the requests from a type-as-you-go input field like this. First, a request is not sent until at least 3 characters have been typed. Second, we debounce the calls to a 250ms delay. With debouncing, we can prevent a request being sent every time a letter is typed. It will wait until the user has stopped typing for 250ms. 

We use a library for this in our new JS code, but a simple implementation can be seen here: https://www.freecodecamp.org/news/javascript-debounce-example/

Typically, I think all the debounce delays should be the same across the application (250ms). However, if this is a critical performance issue right now, we could increase that delay to 350-500 as a temporary improvement.

Reply:
@NateWr, as far as I could see from testing, this jQuery auto-complete already has a debounce (I didn't check the timeout value, but by watching the fired requests vs the speed I type, it was ok for me...)

About the PHP part, [here's the reasoning](https://www.php.net/manual/en/features.connection-handling.php) I followed: "If you do not tell PHP to ignore a user abort and the user aborts, your script will terminate".

Reply:
Wow, I didn't know about connection aborted handling on the PHP side! It looks like it is off by default, so that an aborted request will still complete. And I guess that we probably want that in most cases (thinking of scheduled tasks, etc). 

Is there a point in the code here where we can check for connection aborted? Most of the cases I have seen involve loops, but here we have a single command (a database request) that is the primary delay. I guess we could check if the connection was aborted before we execute the database request? I wonder if that would catch many cases?

Reply:
The [default is to abort](https://www.php.net/manual/en/misc.configuration.php#ini.ignore-user-abort), so whenever we're going to run a long loop, there should be a `set_time_limit(0)` + `ignore_user_abort(true)`.

But I've never really tested it before... Today I've decided to check it out and... It's not worth to play with it, the detection mechanism depends on sending output data to the webserver (which may also fail due to response buffer) 😭

Reply:
From the description at https://www.php.net/manual/en/function.ignore-user-abort.php it looks like maybe this only applies to command-line usage of PHP?

> When running PHP as a command line script, and the script's tty goes away without the script being terminated then the script will die the next time it tries to write anything, unless enable is set to true

Reply:
It aborts HTTP requests as well :)
I couldn't easily find a way to interrupt aborted requests without resorting to workarounds, so let it be...

---

Coming back to the delay... I just checked and the [default is 300](https://api.jqueryui.com/autocomplete/#option-delay), should I increase to 500? Perhaps we can invoke @mfelczak, to know if the query update is enough to fix the issue, I've checked against an installation with 11K users and it was ok for me.

Reply:
Let's stick to 300ms for the delay unless we're really desperate. I don't think it will make a big difference in real-world usage.